src/entries/popup/pages/settings/customChain/customChain.tsx

```diff
@@ -68,6 +73,25 @@ export function CustomRPC() {
     [customRPCAssetsForChain],
   );
 
+  useEffect(() => {
+    const fetchMetadata = async () => {
+      if (!!chainId && isValidAddress(asset.address)) {
+        const provider = getProvider({ chainId: Number(chainId) });
+        const metadata = await getAssetMetadata({
+          address: asset.address,
+          provider,
+        });
+        setAsset({
+          address: asset.address,
+          symbol: metadata.symbol || INITIAL_ASSET.symbol,
+          decimals: metadata.decimals || INITIAL_ASSET.decimals,
+          name: metadata.name || INITIAL_ASSET.name,
+        });
+      }
+    };
+    fetchMetadata();
+  }, [asset.address, chainId]);
```

Comment:
why not a `useQuery`?

Reply:
oh yes, will use useQuery

Reply:
ok done https://www.loom.com/share/9def5ef7d77949c68736f4fb0b748775